### PR TITLE
Manual scan full page implementation

### PR DESCRIPTION
**Pull request by future-pirate-king**

### CODE REVIEW

File: app/components/file-details/manual-scan/index.ts [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+        delete userRole.id;
+      });
+    }
+
+    const tPleaseEnterVPNDetails = this.intl.t(
+      'modalCard.manual.pleaseEnterVPNDetails'
+    );
+
+    const vpnRequired = this.manualscan?.vpnRequired;
+
```
Comment (Yibaebi) [inline app/components/file-details/manual-scan/index.ts:131]: Remove space here and in line 169.

File: app/components/file-details/manual-scan/request-access/index.ts [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+      await this.ajax.post(url);
+
+      this.notify.success(this.intl.t('accessRequested'));
+
+      await this.args.file.reload();
+    } catch (error) {
+      const err = error as AdapterError;
+      let errMsg = this.intl.t('pleaseTryAgain');
+
+      if (err.errors && err.errors.length) {
```
Comment (Yibaebi) [inline app/components/file-details/manual-scan/request-access/index.ts:39]: Can we use the `parseError` utils here?

File: app/components/file-details/manual-scan/request-form/additional-comments/index.ts [hunk 3]
```diff
@@ -0,0 +1,19 @@
+import Component from '@glimmer/component';
+import FileModel from 'irene/models/file';
+import ManualscanModel from 'irene/models/manualscan';
```
Comment (Yibaebi) [inline app/components/file-details/manual-scan/request-form/additional-comments/index.ts:3]: Import models as `types`.

File: app/components/file-details/manual-scan/request-form/basic-info/index.ts [hunk 4]
```diff
@@ -0,0 +1,53 @@
+import Component from '@glimmer/component';
+import FileModel from 'irene/models/file';
+import ENUMS from 'irene/enums';
+import ManualscanModel from 'irene/models/manualscan';
```
Comment (Yibaebi) [inline app/components/file-details/manual-scan/request-form/basic-info/index.ts:4]: Add `type` to imports.

File: app/components/file-details/manual-scan/request-form/login-details/index.ts [hunk 5]
```diff
@@ -0,0 +1,122 @@
+import { action } from '@ember/object';
+import { inject as service } from '@ember/service';
+import { isEmpty } from '@ember/utils';
+import Component from '@glimmer/component';
+import { tracked } from '@glimmer/tracking';
+import IntlService from 'ember-intl/services/intl';
+import FileModel from 'irene/models/file';
+import ManualscanModel, { UserRole } from 'irene/models/manualscan';
```
Comment (Yibaebi) [inline app/components/file-details/manual-scan/request-form/login-details/index.ts:8]: Add `type` to imports.

File: app/components/file-details/manual-scan/request-form/login-details/index.ts [hunk 6]
```diff
@@ -0,0 +1,122 @@
...
+        textAlign: 'center',
+      },
+    ];
+  }
+
+  get allUserRoles() {
+    return (
+      this.args.manualscan?.userRoles.map((role, i) => {
+        role.id = i + 1;
+        return role;
```
Comment (Yibaebi) [inline app/components/file-details/manual-scan/request-form/login-details/index.ts:49]: Add space before return.

**Review by Yibaebi — CHANGES_REQUESTED** [review]

File: app/components/file-details/manual-scan/index.ts [hunk 7]
```diff
@@ -0,0 +1,199 @@
...
+    };
+
+    const tPleaseEnterUserRoles = this.intl.t(
+      'modalCard.manual.pleaseEnterUserRoles'
+    );
+
+    const loginRequired = this.manualscan?.loginRequired;
+    const userRoles = this.manualscan?.userRoles;
+
+    if (loginRequired) {
```
[the remark this answers is not in the document]
Comment (future-pirate-king) [reply app/components/file-details/manual-scan/index.ts:114]: done

File: app/components/file-details/manual-scan/index.ts (hunk 1 above)
Comment (future-pirate-king) [reply app/components/file-details/manual-scan/index.ts:131]: done

File: app/components/file-details/manual-scan/request-form/additional-comments/index.ts (hunk 3 above)
Comment (future-pirate-king) [reply app/components/file-details/manual-scan/request-form/additional-comments/index.ts:3]: done

File: app/components/file-details/manual-scan/request-form/basic-info/index.ts (hunk 4 above)
Comment (future-pirate-king) [reply app/components/file-details/manual-scan/request-form/basic-info/index.ts:4]: done

File: app/components/file-details/manual-scan/request-form/login-details/index.ts (hunk 5 above)
Comment (future-pirate-king) [reply app/components/file-details/manual-scan/request-form/login-details/index.ts:8]: done

File: app/components/file-details/manual-scan/request-form/login-details/index.ts (hunk 6 above)
Comment (future-pirate-king) [reply app/components/file-details/manual-scan/request-form/login-details/index.ts:49]: done

File: app/components/file-details/manual-scan/request-access/index.ts (hunk 2 above)
Comment (future-pirate-king) [reply app/components/file-details/manual-scan/request-access/index.ts]: done

**Review by Yibaebi — APPROVED** [review]

**Pull request merged**